### PR TITLE
Changes protobuf

### PR DESCRIPTION
**Pull request by morucci**

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -2,6 +2,7 @@
 # SPDX-License-Identifier: AGPL-3.0-or-later
 
 MESSAGES = monocle/config.proto monocle/search.proto monocle/task_data.proto
+MESSAGES2 = monocle/change.proto
```
Comment (TristanCacqueray) [inline Makefile:5]:
```suggestion
BACKEND_ONLY = monocle/change.proto
```

**Review by TristanCacqueray — COMMENTED** [review]
Perhaps using a common message would be easier to manage, e.g.:

```
message Document {
  string id = 1;
  google.protobuf.Timestamp created_at = 2;
  [... common attribute]
  oneof type {
    Change change;
    ChangeCreated changeCreated;
    ChangeCommented changeCommented;
    [...]
  }
}
```

Then specific field could be provided like so:

```
message ChangeCreated {
  google.protobuf.Timestamp on_created_at = 1;
  [...]
}

message Change {
  int32 commit_count = 1;
  [...]
}
```

**Comment by TristanCacqueray** [comment]
This reminds me a lot what i did for gerrit: hackage.haskell.org/package/gerrit/docs/Gerrit-Event.html :-)

**Comment by morucci** [comment]
> Perhaps using a common message would be easier to manage, e.g.:
> 
> ```
> message Document {
>   string id = 1;
>   google.protobuf.Timestamp created_at = 2;
>   [... common attribute]
>   oneof type {
>     Change change;
>     ChangeCreated changeCreated;
>     ChangeCommented changeCommented;
>     [...]
>   }
> }
> ```
> 
> Then specific field could be provided like so:
> 
> ```
> message ChangeCreated {
>   google.protobuf.Timestamp on_created_at = 1;
>   [...]
> }
> 
> message Change {
>   int32 commit_count = 1;
>   [...]
> }
> ```

Yes that is a good idea I'll go with you proposal.

**Review by TristanCacqueray — APPROVED** [review]

**Pull request merged**